### PR TITLE
Do some Kubectl optimizations suggested by the golangci linter

### PR DESCRIPTION
**Pull request by i02sopop**
**What type of PR is this?**
/kind cleanup
/sig cli

**What this PR does / why we need it**:
The tool golangci-lint gives a bunch of warnings. This PR solves the easier/less controversial ones, so the code is simpler and a little bit more optimal, since it removes some `if` conditions.

**Special notes for your reviewer**:
This is my first PR, so any advice is more than welcome :).

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/repos_generated.bzl
Comment (draveness) [inline staging/repos_generated.bzl]: What does the ci linter complain about this? Others LGTM.

Comment (i02sopop) [reply staging/repos_generated.bzl]: This one was suggested by the `hack/verify-bazel.sh` script, if I remember well, not by the linter. The linter only checks the `.go` files.

**Comment by draveness** [comment]
Please squash the commits into one.

File: staging/repos_generated.bzl
Comment (i02sopop) [reply staging/repos_generated.bzl]: The `verify` test have failed, and it says that I should run `./hack/update-bazel.sh`, but for me it generates the changes commited, should I roll back those changes? Or should I change the test?

Comment (draveness) [reply staging/repos_generated.bzl]: I think you should revert the change of this file.

**Comment by dims** [comment]
/milestone v1.16

we are past the code freeze, so let's line this up for 1.16, please feel free to talk to the assignees when the 1.15 rush is over in a week or two.

**Comment by i02sopop** [comment]
Any updates?

**Comment by draveness** [comment]
> Any updates?

Please squash the commits into one, and ping the approver for reviews.

**Comment by i02sopop** [comment]
@juanvallejo @deads2k Can you review the PR?

**Comment by i02sopop** [comment]
@seans3 Any thoughts?

**Comment by i02sopop** [comment]
@seans3 @juanvallejo @deads2k Can anyone review this PR?

**Comment by ttousai** [comment]
Hello! I'm a bug triage shadow for the 1.16 release cycle and considering this issue is tagged for 1.16, I'd like to check its status. Code freeze is starting on August 29th (about 3 days from now), which means that this PR must merged by then.

Do we still target this issue to be fixed for 1.16?

**Comment by i02sopop** [comment]
From my side yes, I think all the work is done for this issue, but it needs an approval, which I've been waiting three weeks since I've done the last changes to this PR.

**Comment by i02sopop** [comment]
Can anyone review this PR? I don't know why it's taking so long to review, it's not so complicated.

**Comment by i02sopop** [comment]
@juanvallejo @draveness @deads2k Can you please review this PR?

**Comment by smourapina** [comment]
@i02sopop @juanvallejo @draveness @deads2k Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on November 18. Is this PR still on track for 1.17?

**Comment by i02sopop** [comment]
@smourapina I've been waiting for months to get this PR reviewed without any feedback, I don't know what else I need to do to get this PR reviewed :cry:

**Comment by smourapina** [comment]
Pinging @soltysh @smarterclayton @seans3 as well, as they are listed as assignees: can you please assist in reviewing this PR for the 1.17 milestone?

**Comment by i02sopop** [comment]
Sure

**Comment by i02sopop** [comment]
Rebased with the last commits from master, for what is worth it.

**Review by soltysh — APPROVED** [review]
Good stuff! Thanks a ton!
/lgtm
/approve

**Pull request merged**
